File: java/SingleNumber.java
/*
 * Filename: SingleNumber.java
 * Author: Jun Xu
 * Email: jxu.vimacs@gmail.com
 * Created Time: Thu 02 Jul 2015 08:25:05 PM EDT
 */

public class Solution {
	public int singleNumber(int[] nums) {
		int res = 0;

		for (int n: nums) {
			res ^= n;
		}

		return res;
	}
}

File: cpp/single_number.cpp
/*
 * Filename: single_number.cpp
 * Author: Jun Xu
 * Email: junx@cs.bu.edu
 * Created Time: Sun 15 Nov 2015 08:07:45 PM EST
 * Description: 
 */

#include <iostream>

using namespace std;

class Solution
{
	public:
		int singleNumber(vector<int>& nums)
		{
			int res = 0;

			for (const auto n : nums)
			{
				res ^= n;
			}

			return res;
		}
};
